distributed/shuffle/_limiter.py

```diff
@@ -63,19 +65,20 @@ def empty(self) -> bool:
 
     async def wait_for_available(self) -> None:
         """Block until the counter drops below limit"""
+        if not self.full:
+            return
```

Comment:
I've changed the algorithm to exclude non-blocking calls from the mean duration (both with legacy and new metrics).
IMHO it makes more sense like this - but please discuss.

Reply:
I suspect there might be a large performance penalty, but have we considered looking at calculating histograms from (sampled) data points? I'd generally be interested in the distribution of the wait times _including_ non-blocking calls. If that doesn't work, I _think_ I'm more interested in including non-blocking calls here or at least counting them as well. Knowing how long we wait on average (including non-blocking calls) seems to be more relevant to me than knowing how long we wait on average _if we wait_, not knowing anything about how often we wait or don't wait.